kernel/cClosure.ml

```diff
@@ -1501,8 +1501,8 @@ let rec knr info tab m stk =
              | Some m ->
              kni info tab m s
              | None ->
-               let f = {mark = Whnf; term = FFlex (ConstKey c)} in
-               let m = {mark = Whnf; term = FApp(f,args)} in
+               let f = {mark = Cstr; term = FFlex (ConstKey c)} in
+               let m = {mark = Cstr; term = FApp(f,args)} in
```

Comment:
I think this is actually dead code, the red_prim -> None case only happens when we get a NativeDestKO but that's only raised by get_int/get_float (for some reason get_parray uses Not_found).
I think those 3 raises should be assert false and this None branch too.
(red_prim can still return option when used by cbv but not in cclosure reductionops or cbn AFAICT)

Reply:
What happens for under-applied primitive functions?

Reply:
check_native_args in the fflex delta case rejects

Reply:
But can't this happen by zupdate side-effects?

Reply:
What's "this"?
We are in the case where progressing the zprimitive from the stack leaves `nargs` empty, IIUC this guarantees that all the args the primitive looks at are reduced to native values.

Reply:
OK, I'm convinced, let's put an assertion there.